### PR TITLE
cov-mode tests

### PR DESCRIPTION
**Pull request by theldoria**
I added tests for the cov-mode enable and disable and for the overlays.

### CODE REVIEW

**Comment by AdamNiederer** [comment]
Tests look good. Thanks.

File: cov.el [hunk 1]
```diff
@@ -171,7 +171,7 @@ The function iterates over `cov-coverage-file-path' for path candidates or locat
 
 If `cov-coverage-file' is non nil, the value of that variable is returned. Otherwise `cov--locate-coverage' is called."
   (or cov-coverage-file
-      (setq cov-coverage-file (cov--locate-coverage (f-this-file)))))
+      (setq cov-coverage-file (cov--locate-coverage (buffer-file-name)))))
```
Comment (theldoria) [inline cov.el]: That was required, because `f-this-file`  is confused by `ert-runner`.

File: cov.el [hunk 2]
```diff
@@ -240,10 +238,10 @@ code's execution frequency"
   (let ((gcov (cov--coverage)))
     (if gcov
         (let* ((lines (mapcar 'cov--parse (cov--read (car gcov))))
-               (max (reduce 'max (cons 0 (mapcar 'cl-second lines)))))
+               (max (cl-reduce 'max (cons 0 (mapcar 'cl-second lines)))))
```
Comment (theldoria) [inline cov.el]: Function `reduce` was not known using `ert-runner`, bus it is an alias to `cl-reduce`, which is known.

File: cov.el [hunk 3]
```diff
@@ -266,6 +264,7 @@ code's execution frequency"
 
 (defun cov-turn-on ()
   "Turn on cov-mode."
+  (cov-clear-overlays)
```
Comment (theldoria) [inline cov.el]: Tests unveiled that, if you re-enable the mode the overlays are re-created. Clearing them beforehand solves this.

**Pull request merged**